### PR TITLE
Small fix around

### PR DESCRIPTION
**Pull request by alessiofumagalli**

### CODE REVIEW

**Comment by alessiofumagalli** [comment]
only test cases related to thm fails as in the develop branch
codacy complains that a function (in the coarsening module) is too long, but I cannot do much with that

**Comment by IvarStefansson** [comment]
Could you provide a short description of this branch?

**Comment by alessiofumagalli** [comment]
This branch changes slightly some of the interfaces of the internal functions of the coarsening module.
A couple of fixes in the visualization modules (plot_grid and vtk export)
change the tests to be compliant with the new interfaces.

Should be an easy merge :P

File: src/porepy/grids/coarsening.py [hunk 1]
```diff
@@ -511,6 +512,12 @@ def create_partition(A, seeds=None, **kwargs):
     cdepth = int(kwargs.get("cdepth", 2))
     epsilon = kwargs.get("epsilon", 0.25)
 
+    # NOTE: Extract the higher dimensional grids, we suppose it is unique
+    if isinstance(g, grid_bucket.GridBucket):
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: fixed

File: src/porepy/grids/coarsening.py [hunk 2]
```diff
@@ -511,6 +512,12 @@ def create_partition(A, seeds=None, **kwargs):
...
+        g_high = g.get_grids(lambda g_: g_.dim == g.dim_max())[0]
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: probably something old, fixed now

File: src/porepy/grids/coarsening.py [hunk 3]
```diff
@@ -679,7 +687,8 @@ def create_partition(A, seeds=None, **kwargs):
             vals += sps.csr_matrix((nv, (nc, np.repeat(mi, len(nc)))), shape=(Nc, NC))
 
     coarse, fine = primal.tocsr().nonzero()
-    return coarse[np.argsort(fine)]
+    partition = {g_high: (g_high.copy(), coarse[np.argsort(fine)])}
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: done

File: src/porepy/grids/coarsening.py [hunk 4]
```diff
@@ -541,7 +548,8 @@ def create_partition(A, seeds=None, **kwargs):
             row = np.hstack([STold.rows[r] for r in rowj])
             ST[j, np.concatenate((rowj, row))] = True
 
-    del STold
+    if "STold" in locals():
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: not need to keep that variable, I can remove the line if you want or just leave it. anyway this function is quite inefficient, and it's used only to make a point.

File: src/porepy/grids/coarsening.py [hunk 5]
```diff
@@ -242,6 +242,7 @@ def generate_coarse_grid_gb(gb, subdiv):
     for g, (_, partition) in subdiv.items():
 
         # Construct the coarse grids
+        #part = next(iter(partition.values()))[1]
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: done

File: src/porepy/utils/half_space.py [hunk 6]
```diff
@@ -105,7 +105,11 @@ def half_space_pt(n, x0, pts, recompute=True):
         (0, None),
         (0, None),
     )
-    res = opt.linprog(c, A_ub, b_ub, bounds=bounds)
+    try:
+        res = opt.linprog(c, A_ub, b_ub, bounds=bounds)
+    except ValueError:
+        # it fails to compute the internal point
+        return np.array([np.nan, np.nan, np.nan])
```
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/utils/half_space.py]: this function is called only in the half_space.py module. In this case we are computing the point that make the cell star-shaped, if the point is not found (the cell is not star-shaped) I need to communicate this to the user. what to do? I can simply raise an exception

**Comment by alessiofumagalli** [comment]
@keileg can you give me some updates?

File: src/porepy/utils/half_space.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/utils/half_space.py]: Yes, please raise an exception.

File: src/porepy/grids/coarsening.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (keileg) [reply src/porepy/grids/coarsening.py]: Please delete the line then.

**Comment by keileg** [comment]
Inline above. 

Sorry for the delay, my workflow with GitHub is still messed up after Google killed inbox.

File: src/porepy/utils/half_space.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/utils/half_space.py]: okay

File: src/porepy/grids/coarsening.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (alessiofumagalli) [reply src/porepy/grids/coarsening.py]: okay

**Comment by alessiofumagalli** [comment]
Okay, @keileg  I should have included all of your suggestions. Let me know what do you think.

**Pull request merged**